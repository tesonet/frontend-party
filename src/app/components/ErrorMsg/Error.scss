$error_color: #dd5a5a;
$error_color_text: #f9f9f9;

@mixin errorAnimation($time) {
  -webkit-animation: fade-in $time ease-in forwards;
  -moz-animation: fade-in $time ease-in forwards;
  -ms-animation: fade-in $time ease-in forwards;
  -o-animation: fade-in $time ease-in forwards;
  animation: fade-in $time ease-in forwards;
}

@mixin errorAnimationTransformation() {
  0% {
    top: -30px;
    opacity: 0.1;
  }
  15% {
    top: -20px;
    opacity: 0.2;
  }
  30% {
    top: -10px;
    opacity: 0.4;
  }
  50% {
    top: 0;
    opacity: 0.6;
  }
  100% {
    top: 0px;
    opacity: 1;
  }
}

.error-exit {
  position: absolute;
  top: 0.8em;
  right: 1.5em;
  cursor: pointer;
}

.error-hidden {
  display: none;
}

.error-container {
  position: absolute;
  width: 100%;
  top: 0;
  left: 0;
  padding: 0.8em 0 0.5em;

  display: flex;
  justify-content: center;
  align-items: center;

  color: $error_color_text;
  background: $error_color;

  @include errorAnimation(0.3s);
}

@keyframes fade-in {
  @include errorAnimationTransformation();
}

@-webkit-keyframes fade-in {
  @include errorAnimationTransformation();
}

@-moz-keyframes fade-in {
  @include errorAnimationTransformation();
}

@-ms-keyframes fade-in {
  @include errorAnimationTransformation();
}

@-o-keyframes fade-in {
  @include errorAnimationTransformation();
}
