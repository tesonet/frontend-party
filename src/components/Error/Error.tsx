import React from "react";
import styled from "styled-components";

interface ErrorBoxProps {
  message: string | null;
}

export const ErrorContainer = styled.div<{ error: boolean }>`
  border: 1px solid rgb(209, 65, 54);
  color: #fff;
  padding: 1rem;
  background: rgba(209, 65, 54, 0.7);
  border-radius: 0.3rem;
  visibility: ${props => (props.error ? "visible" : "hidden")};
`;

export const ErrorBox: React.FC<ErrorBoxProps> = ({ message }) => {
  return <ErrorContainer error={!!message}>{message}</ErrorContainer>;
};
