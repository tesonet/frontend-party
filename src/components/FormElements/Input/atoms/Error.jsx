import styled from 'styled-components';
import { color } from 'styled-system';

import { BodyText } from 'components/Typography';

export default styled(BodyText)`
  font-size: 12px;
  padding-top: 10px;
  padding-left: 20px;

  ${color}
`;
