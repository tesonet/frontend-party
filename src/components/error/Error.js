import React from 'react';
import PropTypes from 'prop-types';
import style from './Error.scss';

const Error = ({ message }) => (
  message ? (<div className={style.error}>{message}</div>) : null
);

Error.defaultProps = {
  message: '',
};

Error.propTypes = {
  message: PropTypes.string,
};

export default Error;
