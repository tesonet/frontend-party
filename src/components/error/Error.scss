.error {
  color: var(--danger);
  margin: -20px 0 20px 0;
  width: 100%;
  font: 600 16px 'Roboto', sans-serif;
  text-align: center;
}
