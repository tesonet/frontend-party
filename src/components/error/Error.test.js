import React from 'react';
import Enzyme, { render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Error from './Error';

Enzyme.configure({ adapter: new Adapter() });

test('Error renders without crashing', () => {
  shallow(<Error />);
});

test('Error renders correct message', () => {
  const message = 'testing';
  const e = render(<Error message={message} />);
  expect(e.html()).toMatchSnapshot();
});
