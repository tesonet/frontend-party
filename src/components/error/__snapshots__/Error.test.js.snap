// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correct message 1`] = `"testing"`;
